src/content/en/fundamentals/engage-and-retain/push-notifications/subscribing-a-user.md

```diff
@@ -0,0 +1,396 @@
+          )
+        };
+
+        return registration.pushManager.subscribe(subscribeOptions);
+      })
+      .then(function(pushSubscription) {
+        console.log('Received PushSubscription: ', JSON.stringify(pushSubscription));
+        return pushSubscription;
+      });
+    }
+
+When calling the `subscribe()` method, we pass in an *options* object, which consists of both
+required and optional parameters.
+
+Lets look at all the options we can pass in.
+
+### userVisibleOnly Options
+
+When push was first added to browsers, there was uncertainty about whether developers should be
+able to send a push message and not show a notification. This is commonly referred to as silent
+push, due to the user not knowing that something had happened in the background.
+
+The concern was that developers could do nasty things like track a user's location on an
+ongoing basis without the user knowing.
+
+To avoid this scenario and to give spec authors time to consider how best to support this
+feature, the `userVisibleOnly` option was added and passing in a value of `true` is a symbolic
+agreement with the browser that the web app will show a notification every time a push is
+received (i.e. no silent push).
+
+At the moment you **must** pass in a value of `true`. If you don't include the
+`userVisibleOnly` key or pass in `false` you'll get the following error:
+
+> Chrome currently only supports the Push API for subscriptions that will result in
+user-visible messages. You can indicate this by calling
+`pushManager.subscribe({userVisibleOnly: true})` instead. See
+[https://goo.gl/yqv4Q4](https://goo.gl/yqv4Q4) for more details.
+
+It's currently looking like blanket silent push will never be implemented in Chrome. Instead,
+spec authors are exploring the notion of a budget API which will be allow web apps a certain
+number of silent push messages based on the usage of a web app.
+
+
+
+### applicationServerKey Option
+
+We briefly mentioned the notion "application server keys" in the previous section. "Application
+server keys" are used by a push service to identify the application subscribing a user and
+ensuring that the same application is messaging that user.
+
+Application server keys are a public and private key pair that are unique to your application.
+The private key should be kept a secret to your application and the public key can be shared
+freely.
+
+The `applicationServerKey` option passed into the `subscribe()` call is the applications public
+key. The browser passes this onto a push service when subscribing the user, meaning the push
+service can tie your applications public key to the users `PushSubscription`.
+
+The diagram below illustrates these steps.
+
+1. You web app is loaded in a browser and you call `subscribe()` passing in your public
+application server key.
+1. The browser then makes a network request to a push service who will generate an endpoint,
```

Comment:
For this step:
* Delete 'then'. The numbered steps already cover that bit. 
* Change 'a push service who will generate' to 'a push service generates'.
* Change 'associate this endpoint' to 'associates this endpoint'
* application's not applications
* returns not return